### PR TITLE
add animations to plot_ppc

### PR DESCRIPTION
**Pull request by aloctavodia**
Some examples

`kind=density`
00_animated

`kind=scatter`
01_animated

`kind=cumulative`
02_animated

`kind=density` discrete variables
03_animated

### CODE REVIEW

**Comment by ahartikainen** [comment]
Yes. Like having 3 to 10 samples per frame.

File: arviz/plots/ppcplot.py [hunk 1]
```diff
@@ -432,11 +455,103 @@ def plot_ppc(
...
                 ax.legend([])
 
-    return axes
+    if animated:
+        ani = animation.FuncAnimation(
+            fig,
+            animate,
+            np.arange(0, num_pp_samples),
+            init_func=init,
+            blit=True,
```
[the remark this answers is not in the document]
Comment (ahartikainen) [reply arviz/plots/ppcplot.py]: `blit=True`

Not sure what is the situation now

https://github.com/matplotlib/matplotlib/issues/531

**Comment by aloctavodia** [comment]
Not with the current implementation but it can be added. A friend of mine suggested to have an option to pile-up samples, probably with a lower value of alpha. My concern with these options is if they goes against the purpose of the animation, that is to minimize "deterministic construal errors", but honestly I have to learn more about this and about Hypothetical Outcome Plots.

File: arviz/plots/ppcplot.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (aloctavodia) [reply arviz/plots/ppcplot.py]: So, what we can do here, add a keyword so users can set the value of `blit`?, or we detect the OS and if it is osx we use `TkAgg`? (BTW I don't have access to a machine with osx)

[the remark this answers is not in the document]
Comment (ahartikainen) [reply arviz/plots/ppcplot.py]: I think let users decide. Maybe throw warning on osx?

[the remark this answers is not in the document]
Comment (aloctavodia) [reply arviz/plots/ppcplot.py]: Do you like something like "If you experience problems rendering the animation try turning blit to False or setting the plotting backend to TkAgg".

[the remark this answers is not in the document]
Comment (aloctavodia) [reply arviz/plots/ppcplot.py]: or better "If you experience problems rendering the animation try setting `animation_kwargs({'blit':False}) or setting the plotting backend to TkAgg."

[the remark this answers is not in the document]
Comment (ahartikainen) [reply arviz/plots/ppcplot.py]: Latter

**Comment by aloctavodia** [comment]
@ahartikainen  If you don't mind I prefer to left the N-samples per frame for another PR, after I have the time to do some more reading. I could open an issue.

**Comment by aloctavodia** [comment]
@ahartikainen do you think this is ready to merge? (and further improve in the future)

**Comment by ahartikainen** [comment]
Yes, LGTM, sorry I have missed your question before.

**Pull request merged**